### DISCUSSION TITLE
scripts being automatically saved when closed

### DISCUSSION THREAD

**Discussion by TobyHewing** [dbeaver/dbeaver#34469] [Q&A] [answered]
### Description

If I open a new script / amend an existing script so that an asterisk appears in the title to indicate the file has changed in the past closing the tab would pop up the 'save changes' dialog box but now the tab is closing saving the changes without prompting. Have I inadvertently enabled something?

### DBeaver Version

Community Version 24.1.1.202406231636

### Operating System

macOS Sonoma 14.5

### Database and driver

I'm connecting to both postgres (PostgreSQL standard driver) and BigQuery (simba Google BigQuery driver)

### Steps to reproduce

1. Open a new script
2. Write a query
3. Close the tab by clicking on the x
4. The script is saved without any prompt

**Comment by E1izabeth** [comment, marked as answer]
Hi, @TobyHewing 
We turned on `Auto-save editor on close` by default recently. However, you can always turn off this behavior in settings
![image](https://github.com/dbeaver/dbeaver/assets/28875055/875a22aa-d5ef-4955-9043-91a157d36933)

**Comment by TobyHewing** [comment]
perfect thank you.
